### ISSUE TITLE
Spir-v source view should add a button "Convert to GLSL" to allow human readability.

### ISSUE THREAD

**Issue by pau-baiget** [google/gapid#2614]

**Comment by AWoloszyn** [comment]
I have a preference to use the debugging information that can be embedded in the source to do this. There is an option in some (all?) of the compilers to embed the source in the shader, as well as the command-line arguments needed to compile it.

This is simply because SPIR-V does not have to come from GLSL. It  can come from any number of source languages. If no source is added, we can try spirv-cross, but if an extension, or feature is used that is not supported, it might mis-compile the source, and change the context.

**Comment by ben-clayton** [comment]
I agree that embedded source is the ideal, however it is pretty difficult to do anything with the SPIR-V shader without first converting it to something higher level.
Assuming the user is fully aware of the issues you describe, do you not think providing a automatic conversion to and from GLSL / HLSL would be a very useful feature for experimentation?

Full disclaimer: I proposed this feature to Pau. 😄

**Comment by ben-clayton** [comment]
Note: we already link in SPIR-V cross

**Comment by pau-baiget** [comment]
Changed the title since just adding a convert-to-glsl button may trigger lots of cases where SPIR-V translation to GLSL is not trivial and/or produce wrong output. Instead, adding a Help button that explains where how to get the GLSL translation from an external tool and warning about it not being 100% reliable is the best way to move forward.
